webapp/src/components/CollectionSelect.vue

```diff
@@ -119,23 +126,47 @@ export default {
         loading(false);
       }, debounceTime);
     },
+    IDValidationMessage() {
+      if (this.searchQuery && !/^[a-zA-Z0-9_-]+$/.test(this.searchQuery)) {
+        return "ID can only contain alphanumeric characters, dashes ('-'), and underscores ('_').";
+      }
```

Comment:
```suggestion
      }
      if (/^[._-]/.test(this.searchQuery) | /[._-]$/.test(this.searchQuery)) {
        return "ID cannot start or end with punctuation";
      }
```

Just another case of the item ID validation that was missing. We actually now repeat this code 3/4 times in different modals (with the same typos in the error message, fixed here...), so we should wrap this up into a single utility function for IDs (perhaps with the ability to disable some via bool toggles depending on context). Perhaps you could throw this together after @BenjaminCharmes?